server/conf/routes

```diff
@@ -22,3 +22,5 @@ POST    /register                   controllers.UserController.postRegister
 #Login
 GET     /login                      controllers.UserController.getLogin
 POST    /login                      controllers.UserController.postLogin
+GET     /githublogin                controllers.GitHubUser.getLogin
+GET     /githubback                 controllers.GitHubUser.githubCallback(code: String)
```

Comment:
You can check more details here: https://developer.github.com/v3/oauth/#web-application-flow
1) Probably no. I registered devgym at Github as localhost )
2) Correct. As mentioned before we need to keep a generated random state, so that callback param could match and allow user to proceed with login